### PR TITLE
Improve version info, vsix installation, and update building locally documentation

### PR DESCRIPTION
**Pull request by middlemose**
Additional improvements to the maintenance and documentation of the CI pipeline.

- Improved the vsix installer procedure by auto resolving the latest vsix, added additional logging, and removed sourcing of environment variables.
- Added NuGet package version and vsix version to the sample set build output.
- Updated the documentation for building locally by removing hardcoded version and improved the steps to install the WDK vsix extension from Visual Studio.

### CODE REVIEW

**Comment by middlemose** [comment]
Install vsix output:
```
Run .\.github\scripts\Install-Vsix.ps1 -optimize:$true
**********************************************************************
** Visual Studio 2022 Developer PowerShell v17.10.3
** Copyright (c) 2022 Microsoft Corporation
**********************************************************************
---> Downloading vsix version: 10.0.2[6](https://github.com/microsoft/Windows-driver-samples/actions/runs/9668696224/job/26673422480?pr=1190#step:3:7)100.1
Downloaded VSIX Version: 10.0.26100.1
<--- Download complete
---> Configuring build environment...
Installed VSIX Version: 10.0.26100.1
<--- Configuration complete
```

**Comment by middlemose** [comment]
Build info output:
```
Build Environment:          GitHub
Build Number:               26100
Nuget Package Version:      10.0.26100.1
WDK VSIX Version:           10.0.26100.1
Samples:                    132
Configurations:             1 (Release)
Platforms:                  1 (x64)
InfVerif_AdditionalOptions: /samples
Combinations:               132
LogicalProcessors:          4
ThrottleFactor:             5
ThrottleLimit:              20
WDS_WipeOutputs:            
Disk Remaining (GB):        144.584812164307
T: Combinations
B: Built
R: Build is running currently
P: Build is pending an available build slot
S: Built and result was 'Succeeded'
E: Built and result was 'Excluded'
U: Built and result was 'Unsupported' (Platform and Configuration combination)
F: Built and result was 'Failed'
O: Built and result was 'Sporadic'
Building all combinations...
```

File: Build-SampleSet.ps1 [hunk 1]
```diff
@@ -163,8 +164,23 @@ $jresult = @{
 
 $SolutionsTotal = $sampleSet.Count * $Configurations.Count * $Platforms.Count
 
+# Find vsix version either from env variabel or from packages
+$vsix_version = $env:SAMPLES_VSIX_VERSION
```
Comment (JakobL-MSFT) [inline Build-SampleSet.ps1:168]: Why do the version conditionally from variable or from file(s)?  Why not always from file(s)?

File: .github/scripts/Install-Vsix.ps1 [hunk 2]
```diff
@@ -12,34 +12,42 @@ param(
...
 
-# source environment variables
-. .\Env-Vars.ps1
+# Set local version variable
+$version = ([regex]'(\d+\.)(\d+\.)(\d+\.)(\d+)').Matches($uri).Value
 
-$version = $env:SAMPLES_VSIX_VERSION
-$uri = $env:SAMPLES_VSIX_URI
+# Set github environment variable for vsix version
+"SAMPLES_VSIX_VERSION=$version" | Out-File -FilePath "$env:GITHUB_ENV" -Append
```
Comment (JakobL-MSFT) [inline .github/scripts/Install-Vsix.ps1:27]: Hmm.  I need to understand this better.  Will drop by.  Seems convoluted to me.

**Review by JakobL-MSFT — DISMISSED** [review]

File: .github/scripts/Install-Vsix.ps1 [hunk 3]
```diff
@@ -12,34 +12,42 @@ param(
...
-    "WDK Vsix Version: $installed"
+    "WDK Vsix Version: $(ls "${env:ProgramData}\Microsoft\VisualStudio\Packages\Microsoft.Windows.DriverKit,version=*" | Select -ExpandProperty Name)"
 }
 
 function TestWdkVsix {
     Test-Path "${env:ProgramData}\Microsoft\VisualStudio\Packages\Microsoft.Windows.DriverKit,version=$version"
 }
 
+# NOTE: The '$optimize' code path examines the '.vsixmanifest' when downloaded and then examines it again (in the 'MSBuild' directory) once 
+#       the necessary extension files are copied.
```
[the remark this answers is not in the document]
Comment (middlemose) [reply .github/scripts/Install-Vsix.ps1]: I removed this. No more optimize which I believe you suggested before.

File: .github/scripts/Install-Vsix.ps1 [hunk 4]
```diff
@@ -12,34 +12,42 @@ param(
...
-    $installed = ls "${env:ProgramData}\Microsoft\VisualStudio\Packages\Microsoft.Windows.DriverKit,version=*" | Select -ExpandProperty Name
-    "WDK Vsix Version: $installed"
+    "WDK Vsix Version: $(ls "${env:ProgramData}\Microsoft\VisualStudio\Packages\Microsoft.Windows.DriverKit,version=*" | Select -ExpandProperty Name)"
 }
 
 function TestWdkVsix {
     Test-Path "${env:ProgramData}\Microsoft\VisualStudio\Packages\Microsoft.Windows.DriverKit,version=$version"
 }
 
+# NOTE: The '$optimize' code path examines the '.vsixmanifest' when downloaded and then examines it again (in the 'MSBuild' directory) once
```
[the remark this answers is not in the document]
Comment (middlemose) [reply .github/scripts/Install-Vsix.ps1]: I removed this. No more optimize which I believe you suggested before.

File: .github/scripts/Install-Vsix.ps1 (hunk 2 above)
Comment (middlemose) [reply .github/scripts/Install-Vsix.ps1]: This is easier to deal with in our pipeline when environment is GitHub.

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions?tool=powershell#setting-an-environment-variable

File: Build-SampleSet.ps1 (hunk 1 above)
Comment (middlemose) [reply Build-SampleSet.ps1]: I moved this to the block `if ($env:GITHUB_REPOSITORY) {` as fair compromise instead dealing with files.

**Comment by middlemose** [comment]
Building samples locally passed.
```
Build Environment:          NuGet
Build Number:               26100
Nuget Package Version:      10.0.26100.1
WDK VSIX Version:           10.0.26100.1
Samples:                    132
Configurations:             2 (Debug Release)
Platforms:                  2 (x64 arm64)
InfVerif_AdditionalOptions: /samples
Combinations:               528
LogicalProcessors:          20
ThrottleFactor:             5
ThrottleLimit:              100
WDS_WipeOutputs:
Disk Remaining (GB):        742.531757354736
```
```
Built all combinations.

Elapsed time:         13 minutes, 17 seconds.
Disk Remaining (GB):  734.718753814697
Samples:              132
Configurations:       2 (Debug Release)
Platforms:            2 (x64 arm64)
Combinations:         528
Succeeded:            526
Excluded:             0
Unsupported:          2
Failed:               0
Sporadic:             0
Log files directory:  D:\my_stuff\Windows-driver-samples\_logs
Overview report:      D:\my_stuff\Windows-driver-samples\_logs\_overview.htm
```
[_overview.csv](https://github.com/user-attachments/files/16021783/_overview.csv)

**Review by NeoAdonis — APPROVED** [review]

**Pull request merged**